test/functional/tests/fault_injection/test_fault_injection_opencas_load.py

```diff
@@ -0,0 +1,152 @@
+from core.test_run import TestRun
+from test_tools import fs_utils
+from test_tools.dd import Dd
+from test_tools.disk_utils import Filesystem
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+test_file_path = f"{mount_point}/test_file"
+
+
+@pytest.mark.parametrize("cache_mode", [CacheMode.WB, CacheMode.WO])
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_stop_no_flush_load_cache(cache_mode, filesystem):
+    """
+        title: Test to check that 'stop --no-data-flush' command works correctly.
+        description: |
+          Negative test of the ability of CAS to load unflushed cache on core device
+          with and without filesystem. Test uses write-back and write-only cache modes.
+        pass_criteria:
+          - No system crash while load cache.
+          - Starting cache without loading metadata fails.
+          - Starting cache with loading metadata finishes with success.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_part = core_dev.partitions[0]
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+
+    with TestRun.step("Change cleaning policy to NOP."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step(f"Add core with {filesystem.name} filesystem to cache and mount it."):
+        core_part.create_filesystem(filesystem)
+        core = cache.add_core(core_part)
+        core.mount(mount_point)
+
+    with TestRun.step(f"Create test file in mount point of exported object and check its md5 sum."):
+        test_file = fs_utils.create_random_test_file(test_file_path, Size(48, Unit.MebiByte))
+        test_file_md5_before = test_file.md5sum()
+
+    with TestRun.step("Count dirty blocks on exported object."):
+        dirty_blocks_before = cache.get_dirty_blocks()
+
+    with TestRun.step("Unmount exported object."):
+        core.unmount()
+
+    with TestRun.step("Stop cache with option '--no-data-flush'."):
+        cache.stop(no_data_flush=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 0:
+            TestRun.fail(f"Expected caches count: 0; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Try to start cache without loading metadata."):
+        output = TestRun.executor.run_expect_fail(cli.start_cmd(
+            cache_dev=str(cache_part.system_path), cache_mode=str(cache_mode),
+            force=False, load=False))
+        cli_messages.check_msg(output, cli_messages.start_wo_load_on_cache_dev_prev_used)
```

Comment:
cli_messages.start_wo_load_on_cache_dev_prev_used - this is wrong expected message, because you passed wrong value of cache mode parameter. The message should be like (you should check it, because I paste here the example output):
"Error inserting cache 1
Old metadata found on device.
Please load cache metadata using --load option or use --force to
 discard on-disk metadata and start fresh cache instance.

Error occurred, please see syslog (/var/log/messages) for details. 
"